src/mem_table.rs

```diff
@@ -36,7 +40,52 @@ pub(crate) struct ImmutableWal {
 
 type MemTableRange<'a> = Range<'a, Bytes, (Bound<Bytes>, Bound<Bytes>), Bytes, ValueWithAttributes>;
 
-pub struct MemTableIterator<'a>(MemTableRange<'a>);
+pub(crate) struct MemTableIterator<'a>(MemTableRange<'a>);
+
+pub(crate) struct VecDequeKeyValueIterator {
+    records: VecDeque<RowEntry>,
+}
+
+impl VecDequeKeyValueIterator {
+    pub(crate) async fn from_range(
+        tables: VecDeque<Arc<KVTable>>,
+        range: BytesRange,
+    ) -> Result<Self, SlateDBError> {
+        let mut memtable_iters = VecDeque::new();
+        for table in &tables {
+            memtable_iters.push_back(table.range_iter(range.clone()));
+        }
+
+        let mut merge_iter = MergeIterator::new(memtable_iters).await?;
+        let mut records = VecDeque::new();
+        loop {
```

Comment:
Basically I couldn't come up with a nice lazy approach which didn't rely on self-referencing data structures (such as through ouroborus which is used by minilsm). I discussed with @rodesai and it seemed simpler for the initial work to filter memory tables up-front. 

I'm definitely open to suggestions here. It's probably possible to refactor the API so that `scan()` returns some kind of result struct which can then be used to build the iterator. For example:
```rust
let scan_result = db.scan(..).unwrap();
let scan_iter = scan_result.iter();
while Some(record) = iter.next().await? {
...
}
```
I think this would work because the iterator can refer to the data structures in the scan result, but it looks a little clumsier.

Reply:
Yea, let's keep it simple. If it gets in the way we can revisit.